### PR TITLE
Reintroduce whitespace control parsing fix with legacy override

### LINKED ISSUE #892: Unary minus operator at beginning of expression is ignored
Adding parenthesis around an expression with a unary sub-expression, changes the output incorrectly.

`{{ (-10 | abs + 4)}}  is -6`
`{{ -10 | abs + 4 }}  is 14`

In the former example, the unary operation is being applied last, whereas in the latter example, the unary operation is being applied before the `abs` filter (as i would expect)

### PR DESCRIPTION
**Pull request by tkindy**
Closes #892 
Depends on #902 

This PR is basically #896 but with the changes disabled by default. They can be enabled by setting a new flag on `LegacyOverrides` as part of the `JinjavaConfig`.

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/tree/parse/Token.java [hunk 1]
```diff
@@ -83,6 +84,37 @@ public void setRightTrimAfterEnd(boolean rightTrimAfterEnd) {
...
+   * @return the content stripped of any whitespace control characters.
+   */
+  protected final String handleTrim(String unwrapped) {
+    boolean parseWhitespaceControlStrictly =
+      JinjavaInterpreter.getCurrent() != null &&
+      JinjavaInterpreter
+        .getCurrent()
+        .getConfig()
+        .getLegacyOverrides()
+        .isParseWhitespaceControlStrictly();
```
Comment (jasmith-hs) [inline src/main/java/com/hubspot/jinjava/tree/parse/Token.java:100]: Same as https://github.com/HubSpot/jinjava/pull/902#discussion_r932519485 if you like that idea

File: src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.hubspot.jinjava.tree.parse;
+
+public class StrictWhitespaceControlParser implements WhitespaceControlParser {
+
+  @Override
+  public boolean hasLeftTrim(String unwrapped) {
+    return unwrapped.startsWith("-");
```
Comment (jasmith-hs) [inline src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java:7]: Might as well have `unwrapped.charAt(0) == '-'`, which is faster

File: src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  @Override
+  public String stripLeft(String unwrapped) {
+    return unwrapped.substring(1);
+  }
+
+  @Override
+  public boolean hasRightTrim(String unwrapped) {
+    return unwrapped.endsWith("-");
```
Comment (jasmith-hs) [inline src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java:17]: And this could be `unwrapped.charAt(unwrapped.length() - 1) == '-'`

**Review by jasmith-hs — APPROVED** [review]

File: src/main/java/com/hubspot/jinjava/tree/parse/Token.java (hunk 1 above)
Comment (tkindy) [reply src/main/java/com/hubspot/jinjava/tree/parse/Token.java]: Definitely, thanks!

File: src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java (hunk 2 above)
Comment (tkindy) [reply src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java]: Good call, given that this gets called a bunch throughout a whole template.

Comment (tkindy) [reply src/main/java/com/hubspot/jinjava/tree/parse/StrictWhitespaceControlParser.java]: I also added a unit test just to make sure empty expressions don't reach this point without any characters in them (so there's no need to check that the string isn't empty before indexing into it).
